src/crawlee/http_crawler/_http_crawler.py

```diff
@@ -137,5 +150,62 @@ async def _handle_blocked_request(self, context: HttpCrawlingContext) -> AsyncGe
                 and context.session.is_blocked_status_code(status_code=status_code)
             ):
                 raise SessionError(f'Assuming the session is blocked based on HTTP status code {status_code}')
-
+            if blocked_info := self.parser.is_blocked(context.parsed_content):
+                raise SessionError(blocked_info.reason)
         yield context
+
+
+class HttpCrawler(HttpCrawlerGeneric[bytes]):
+    """Specific version of generic HttpCrawlerGeneric.
+
+    It uses dummy parser NoParser. It is not intended to be used in new code, it is backwards compatibility class.
+    In new code either use HttpCrawlerGeneric or other specific children of it - like BeautifulSoupCrawler or
+    ParselCrawler.
```

Comment:
I don't think we will actually remove this one in the future. Could we rephrase this to "hey, unless you're 100% sure what you're doing, you probably want to use BS/Parsel"?

Reply:
I tried to rephrase it.